Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM node:8.4.0-slim
+
+# Prepare
+
+RUN mkdir -p /opt/app
+
+ENV TINI_VERSION v0.16.1
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+RUN chmod +x /tini
+ENTRYPOINT ["/tini", "--"]
+
+ENV PORT 3001
+ENV HOST 0.0.0.0
+EXPOSE $PORT
+
+HEALTHCHECK CMD curl -fs http://$HOST:$PORT/healthz || exit 1
+
+# Install
+
+WORKDIR /opt
+COPY package.json /opt
+COPY yarn.lock /opt
+
+RUN yarn
+
+ENV PATH /opt/node_modules/.bin:$PATH
+
+# Build
+
+WORKDIR /opt/app
+COPY . /opt/app
+
+RUN yarn build && \
+    yarn --production --ignore-scripts --prefer-offline && \
+    yarn cache clean
+
+# Run
+
+ENV NODE_ENV production
```

Comment:
Woudln't it make the code a little more well structured and more readable, if we would bring all the ENV definitions to the top of the file, in one place?

Reply:
Thanks, I'll try to move this one to the top but the reason why it's here in the first place is that if you set node env to production during install phase, you won't be able to install dev depenendenices required during build.

Reply:
Hey, I've restructured the Dockerfile to use the multi stage builds and now all vars are set in the `run` image at the bottom.